components/tidb_query_executors/src/index_scan_executor.rs

```diff
@@ -439,19 +543,47 @@ impl IndexScanExecutorImpl {
         } else {
             remaining.split_at(0)
         };
+        let restore_values = if !remaining.is_empty()
+            && remaining
+                .get(0)
+                .map_or(false, |c| *c == table::INDEX_VALUE_RESTORED_DATA_V5_FLAG)
+        {
+            restored_v5 = true;
+            &remaining[1..]
+        } else if !remaining.is_empty()
+            && remaining
+                .get(0)
+                .map_or(false, |c| *c == table::INDEX_VALUE_RESTORED_DATA_FLAG)
+        {
+            &remaining
+        } else {
+            "".as_bytes()
+        };
```

Comment:
`split_restore_data_from_index_value`?

Reply:
Maybe we can use a function to split all the data like TiDB?

Reply:
Sound good.